File: transportable-udfs-codegen/src/test/resources/outputs/sample-udf-metadata/spark/sources/udfs/spark/OverloadedUDF.scala
package udfs.spark

import java.util
import com.google.common.collect.ImmutableList
import com.linkedin.transport.api.udf.{UDF, TopLevelUDF}
import com.linkedin.transport.spark.{SparkTransportUDF, UDFRegistration, SparkUDF}
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.catalyst.expressions.Expression


case class OverloadedUDF(expressions: Seq[Expression]) extends SparkUDF(expressions) {

  override protected def getTopLevelUdfClass: Class[_ <: TopLevelUDF] = classOf[udfs.OverloadedUDF]

  override protected def getUdfImplementations: util.List[_ <: UDF] = ImmutableList.of(
    new udfs.OverloadedUDFInt(), new udfs.OverloadedUDFString()
  )
}

object OverloadedUDF {

  def register(name: String): SparkTransportUDF = {
    UDFRegistration.register(name, classOf[OverloadedUDF])
  }

  def register(name: String, session: SparkSession): SparkTransportUDF = {
    UDFRegistration.register(name, classOf[OverloadedUDF], session)
  }
}

File: transportable-udfs-codegen/src/test/resources/outputs/sample-udf-metadata/hive/sources/udfs/hive/OverloadedUDF.java
package udfs.hive;

import com.google.common.collect.ImmutableList;
import com.linkedin.transport.api.udf.TopLevelUDF;
import com.linkedin.transport.api.udf.UDF;
import com.linkedin.transport.hive.HiveUDF;
import java.lang.Class;
import java.lang.Override;
import java.util.List;

public class OverloadedUDF extends HiveUDF {
  @Override
  protected Class<? extends TopLevelUDF> getTopLevelUdfClass() {
    return udfs.OverloadedUDF.class;
  }

  @Override
  protected List<? extends UDF> getUdfImplementations() {
    return ImmutableList.of(new udfs.OverloadedUDFInt(), new udfs.OverloadedUDFString());
  }
}
